integreat_cms/cms/views/utils/hix.py

```diff
@@ -0,0 +1,131 @@
+"""
+This file contains functionality to communicate with the textlab api to get the hix-value
+for a given text.
+"""
+from functools import lru_cache
+import json
+import logging
+from urllib.error import HTTPError
+from urllib.request import Request, urlopen
+
+from django.http import JsonResponse
+from django.views.decorators.http import require_POST
+
+from ....api.decorators import json_response
+from ....core.settings import (
+    TEXTLAB_API_URL,
+    TEXTLAB_API_PASSWORD,
+    TEXTLAB_API_USERNAME,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class TextlabClient:
+    """
+    Client for the textlab api.
+    Supports login and hix-score retrieval.
+    """
+
+    def __init__(self, username, password):
+        self.username = username
+        self.password = password
+        self.token = None
+
+        self.login()
+
+    def login(self):
+        """
+        Authorizes for the textlab api. On success, sets the token attribute.
+
+        :raises ValueError: If authentification was not possible
+        """
+        data = {"identifier": self.username, "password": self.password, "ttl": 3600}
+        status, response = self.post_request("/user/login", data)
+        if status != 200:
+            raise ValueError("Could not connect to textlab api")
```

Comment:
I'm not sure if this code can ever be reached - when I enter an invalid password, the login request already throws an exception during server startup: 
```
    from ..views import (
  File "/home/timo/job/integreat/integreat-cms/integreat_cms/cms/views/utils/__init__.py", line 1, in <module>
    from .hix import get_hix_score
  File "/home/timo/job/integreat/integreat-cms/integreat_cms/cms/views/utils/hix.py", line 102, in <module>
    client = TextlabClient(TEXTLAB_API_USERNAME, TEXTLAB_API_PASSWORD)
  File "/home/timo/job/integreat/integreat-cms/integreat_cms/cms/views/utils/hix.py", line 35, in __init__
    self.login()
  File "/home/timo/job/integreat/integreat-cms/integreat_cms/cms/views/utils/hix.py", line 44, in login
    status, response = self.post_request("/user/login", data)
  File "/home/timo/job/integreat/integreat-cms/integreat_cms/cms/views/utils/hix.py", line 97, in post_request
    with urlopen(request) as response:
  File "/usr/lib/python3.9/urllib/request.py", line 214, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/usr/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.9/urllib/request.py", line 561, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.9/urllib/request.py", line 641, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 400: Bad Request
```